graylog2-server/src/main/java/org/graylog2/bootstrap/ServerBootstrap.java

```diff
@@ -115,16 +121,37 @@ protected void beforeStart(TLSProtocolsConfiguration tlsProtocolsConfiguration,
     }
 
     @Override
-    protected void beforeInjectorCreation() {
-        testMongoDbConnection();
+    protected void beforeInjectorCreation(Set<Plugin> plugins) {
+        runPreFlightChecks(plugins);
     }
 
-    // Try to create a single MongoDBConnection, so we can fail early if there is an error
-    private void testMongoDbConnection() {
-        Guice.createInjector(
-                        new NamedConfigParametersModule(jadConfig.getConfigurationBeans()),
-                        new MongoDBModule())
-                .getInstance(MongoConnection.class);
+    private void runPreFlightChecks(Set<Plugin> plugins) {
+        final List<PreflightCheckModule> preflightCheckModules = plugins.stream().map(Plugin::preflightCheckModules)
+                .flatMap(Collection::stream).collect(Collectors.toList());
+
+        final Injector injector = getPreflightInjector(preflightCheckModules);
+
+        // Run server preflight checks
+        injector.getInstance(ServerPreflightCheck.class).runChecks(configuration);
+
+        // Run preflight checks from plugins
+        preflightCheckModules.forEach(m -> m.doCheck(injector));
+    }
```

Comment:
There are currently two ways of running pre-flight checks, the built-in server checks and checks from plugins.

How about introducing some "pre-flight check service" that gets all checks injected, including the server ones?

That way, we can split up the `ServerPreflightCheck` class into smaller modules that we register the same way plugins would do.

Reply:
Great idea :+1: Done :heavy_check_mark: